app-android/tutashared/src/main/java/de/tutao/tutashared/credentials/AuthenticationPrompt.kt

```diff
@@ -48,21 +50,23 @@ class AuthenticationPrompt constructor() {
 		showPrompt(activity, promptInfo, cryptoObject)
 	}
 
-	@Throws(CredentialAuthenticationException::class)
+	@Throws(CredentialAuthenticationException::class, AppLockAuthenticationException::class)
 	private fun showPrompt(
 		activity: FragmentActivity,
 		promptInfo: PromptInfo,
 		cryptoObject: BiometricPrompt.CryptoObject?,
 	) {
 		sem.acquireUninterruptibly()
 		var error: String? = null
+		var errorCode: Int? = null
```

Comment:
It looks good now! I just have a little nitpick.

We should only ever have either both error message and error code or neither. Having one but not another would be an error and we should make it unrepresentable.

I would suggest that we replace these two variables with one of type `Exception` and we construct the error directly in the `onAuthenticationError()` and then just throw it in the end if it's there.

Alternatively we could make a tuple like:

```kt
var error: Pair<Int, String>? = null
```